### PR TITLE
Migrate to SDL2 for more flexibility with Game Controllers

### PR DESCRIPTION
**Pull request by ooeygui**
This change migrates the Windows joystick implementation to SDL2, which enables a vast array of joystick and game controller implementations - including Wireless Xbox One controllers, Accessible controllers and Steering wheels. As implemented, the code will bind to the first available controller when it becomes available.

### CODE REVIEW

File: joy/package.xml [hunk 1]
```diff
@@ -26,7 +26,10 @@
   <depend>roscpp</depend>
   <depend>diagnostic_updater</depend>
   <depend>sensor_msgs</depend>
+  <build_depend>hidapi</build_depend>
```
Comment (seanyen) [inline joy/package.xml]: Is the `hidapi` a dependency for SDL2?

File: joy/src/joy_node_win.cpp [hunk 2]
```diff
@@ -23,76 +23,269 @@ SOFTWARE.
...
-  BOOL bDev1Attached, bDev2Attached; 
- 
-  if((wNumDevs = joyGetNumDevs()) == 0) 
-    return ERR_NODEVICE; 
-
-  bDev1Attached = joyGetPos(JOYSTICKID1,&joyinfo) != JOYERR_UNPLUGGED; 
-  bDev2Attached = wNumDevs == 2 && joyGetPos(JOYSTICKID2,&joyinfo) != JOYERR_UNPLUGGED; 
+// platform
+#if defined(_WIN32)
+    #include <windows.h>
```
Comment (seanyen) [inline joy/src/joy_node_win.cpp]: After a quick glance, I didn't see `Windows` specific code. Do we still need to include `Windows.h` to build?

File: joy/package.xml (hunk 1 above)
Comment (ooeygui) [reply joy/package.xml]: Good catch. This was an intermediate code I was working on before switching to SDL2. I'll update.

File: joy/src/joy_node_win.cpp (hunk 2 above)
Comment (ooeygui) [reply joy/src/joy_node_win.cpp]: You noticed that :-)
This code is fully cross platform; I want to get it make sure it works great using the Windows use case first before proposing Open Robotics to make it the default implementation.  There is an equivalent ROS2 change needed. Perhaps something to bring up at the TSC?

Comment (ooeygui) [reply joy/src/joy_node_win.cpp]: I removed the Windows.h

**Comment by ooeygui** [comment]
Minor update to handle negative values

**Pull request merged**